src/lib/Scenes/Home/Components/ForYou/Components/FairsRail.tsx

```diff
@@ -22,14 +23,24 @@ const IconCarousel = styled.ScrollView`
 
 const TouchableWrapper = styled.View`margin-right: 7;`
 
-class FairsRail extends React.Component<RelayProps, any> {
+interface Props extends RelayProps {
+  registerRailModule?: (module: FairsRail) => void
+  relay?: RelayRefetchProp
+}
+
+export class FairsRail extends Component<Props, any> {
+  componentWillMount() {
+    if (this.props.registerRailModule) {
```

Comment:
I wonder if `<FairsRail ref={this.registerRailModule('fairsRail')} />` would be a better way to do this as `ref` is build in to react components and we could omit this code.

Reply:
Oh interesting, yeah let me try that...

Reply:
@l2succes - so I tried to swap it out, but it looks like the ref points to the wrapper relay container, not the class itself:

<img width="288" alt="screen shot 2017-12-07 at 10 17 52 pm" src="https://user-images.githubusercontent.com/236943/33753548-b2850656-db9c-11e7-8f54-693c16b75672.png">

Thoughts on how to get to the actual class?

Reply:
@damassi Yeah, you have to access `ref.refs.component` to get to the class itself

Reply:
This is related and somewhat counter to my comments [here](https://github.com/artsy/emission/pull/874#discussion_r155533492). I’m ok with either, but we should settle on one.

Reply:
I think it depends on the situation. 
In this particular case, I actually like `registerRailModule` more as it makes the intent clearer, especially since we're registering multiple rails.

But in general, I am leaning towards using `ref` to avoid adding boilerplate code just to get around the fact that we're using a HOC which is wrapping the actual component.

Reply:
I'm open to whatever, but personally I like the the explicit registration as opposed to refs since it seems a little more encapsulated / direct, and having those hooks there for other rail use could be a win in other contexts.

Reply:
@l2succes I agree, but to clarify I was only really talking about this situation with all the home view modules, regardless of type. So it looks like in this situation we’re all onboard for the non-ref callback.